src/components/simple-color-picker/simple-color-picker.component.tsx

```diff
@@ -46,278 +48,285 @@ export interface SimpleColorPickerProps extends ValidationProps, MarginProps {
   value?: string;
 }
 
-export const SimpleColorPicker = (props: SimpleColorPickerProps) => {
-  const {
-    children,
-    error,
-    warning,
-    info,
-    name,
-    legend,
-    onChange,
-    onBlur,
-    onKeyDown,
-    value,
-    isBlurBlocked = false,
-    maxWidth = 300,
-    childWidth = 58,
-    validationOnLegend,
-    required,
-    ...rest
-  } = props;
-
-  const hasProperChildren = useMemo(() => {
-    let hasSimpleColorChildren = true;
-
-    React.Children.toArray(children).forEach((child) => {
-      if (
-        typeof child === "string" ||
-        (React.isValidElement(child) &&
-          (child.type as React.FunctionComponent).displayName !== "SimpleColor")
+export const SimpleColorPicker = React.forwardRef(
+  (props: SimpleColorPickerProps, ref) => {
```

Comment:
This is a question really: since `SimpleColorPicker` is a public component, and we're exposing a `ref` to it, should we document how that's supposed to be used? Especially as we're customising it with `useImperativeHandle` rather than it being a standard DOM ref?

I guess it's supposed to be completely private but maybe consumers might occasionally want to be able to do something with the `gridItemRefs` themselves? In which case we should add some documentation.

We should also ideally add some tests to `SimpleColorPicker` that test this functionality (especially if it's supposed to be public). I know we'll get coverage from the usage in `AdvancedColorPicker`, but some basic tests that attaching a `ref` to `SimpleColorPicker` exposes the properties we want seems sensible to me.

Reply:
Sounds like a good idea, I have added that as a separate commit

Reply:
it looks like you're now exporting the type 👍 , but I don't see either any new tests or any documentation. (I feel the documentation is more important, the tests may not be needed as we have full coverage anyway.)

Reply:
I have added a unit test in SimpleColorPicker. As for documentation - I think that what we have in TypeScript would be enough as it will be converted to PropTypes for js users.